### ISSUE TITLE
Occasional segfaults with default ARC config and DKIM disabled

### ISSUE THREAD

**Issue by flowerysong** [vstakhov/rspamd#1737]
### Classification:

* [x] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [ ] Other bug
* [ ] Feature
* [ ] Enhancement

### Reproducibility:

* [ ] Always
* [X] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:

1.6.1

### Operating system, CPU, memory and environment:

RHEL 7, running on an AWS m4.xlarge EC2 instance

### Description (Please provide a descriptive summary of the issue):

If ARC is not explicitly disabled there are occasional segfaults. After I disabled ARC, the segfaults stopped. I suspect that this is due to the fact that I previously disabled DKIM and something in the ARC code assumes that DKIM is enabled, but I haven't verified this supposition.

### Debugging information:

```
[New LWP 4809]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `rspamd: normal process                       '.
Program terminated with signal 11, Segmentation fault.
#0  rspamd_lru_hash_lookup (hash=0x0, key=0x19d5090, now=now@entry=1499465652) at /usr/src/debug/rspamd-1.6.1/src/libutil/hash.c:323
323		res = g_hash_table_lookup (hash->tbl, key);
#0  rspamd_lru_hash_lookup (hash=0x0, key=0x19d5090, now=now@entry=1499465652) at /usr/src/debug/rspamd-1.6.1/src/libutil/hash.c:323
#1  0x000000000050aacd in lua_dkim_verify_handler (L=0x4099e378) at /usr/src/debug/rspamd-1.6.1/src/plugins/dkim_check.c:1482
#2  0x00007f128601cbd6 in lj_BC_FUNCC () from /lib64/libluajit-5.1.so.2
#3  0x00007f12860602d0 in lua_pcall (L=L@entry=0x4099e378, nargs=nargs@entry=1, nresults=nresults@entry=-1, errfunc=errfunc@entry=1) at lj_api.c:1052
#4  0x0000000000495baa in lua_metric_symbol_callback (task=0x176ab30, ud=0x11c66c0) at /usr/src/debug/rspamd-1.6.1/src/lua/lua_config.c:1086
#5  0x00000000004d6427 in rspamd_symbols_cache_check_symbol (task=task@entry=0x176ab30, item=0x11aa3a8, total_diff=total_diff@entry=0x0, checkpoint=0x1ab68a8, cache=0x1046030) at /usr/src/debug/rspamd-1.6.1/src/libserver/symbols_cache.c:1287
#6  0x00000000004d6899 in rspamd_symbols_cache_check_deps (task=task@entry=0x176ab30, cache=cache@entry=0x1046030, item=item@entry=0x11aa720, checkpoint=checkpoint@entry=0x1ab68a8, recursion=recursion@entry=0, check_only=check_only@entry=0) at /usr/src/debug/rspamd-1.6.1/src/libserver/symbols_cache.c:1395
#7  0x00000000004d84fb in rspamd_symbols_cache_process_symbols (task=task@entry=0x176ab30, cache=0x1046030, stage=stage@entry=16) at /usr/src/debug/rspamd-1.6.1/src/libserver/symbols_cache.c:1693
#8  0x00000000004da937 in rspamd_task_process (task=task@entry=0x176ab30, stages=stages@entry=16383) at /usr/src/debug/rspamd-1.6.1/src/libserver/task.c:720
#9  0x00000000004da80a in rspamd_task_process (task=task@entry=0x176ab30, stages=stages@entry=16383) at /usr/src/debug/rspamd-1.6.1/src/libserver/task.c:818
#10 0x00000000004daa44 in rspamd_task_fin (arg=0x176ab30) at /usr/src/debug/rspamd-1.6.1/src/libserver/task.c:172
#11 0x00000000004c2e3b in rspamd_session_pending (session=session@entry=0x1abb240) at /usr/src/debug/rspamd-1.6.1/src/libserver/events.c:291
#12 0x00000000004c2f75 in rspamd_session_remove_event (session=0x1abb240, fin=<optimized out>, ud=<optimized out>) at /usr/src/debug/rspamd-1.6.1/src/libserver/events.c:226
#13 0x0000000000bc9405 in rdns_process_read (fd=<optimized out>, arg=<optimized out>) at /usr/src/debug/rspamd-1.6.1/contrib/librdns/resolver.c:281
#14 0x00007f1289a67a14 in event_base_loop () from /lib64/libevent-2.0.so.5
#15 0x000000000044b59b in start_worker (worker=0x153b510) at /usr/src/debug/rspamd-1.6.1/src/worker.c:648
#16 0x00000000004e1022 in rspamd_fork_worker (rspamd_main=rspamd_main@entry=0x103cf50, cf=cf@entry=0x10f4880, index=index@entry=1, ev_base=ev_base@entry=0x1521000) at /usr/src/debug/rspamd-1.6.1/src/libserver/worker_util.c:606
#17 0x0000000000449743 in spawn_worker_type (rspamd_main=0x103cf50, ev_base=0x1521000, cf=0x10f4880) at /usr/src/debug/rspamd-1.6.1/src/rspamd.c:523
#18 0x0000000000449cf6 in spawn_workers (rspamd_main=rspamd_main@entry=0x103cf50, ev_base=ev_base@entry=0x1521000) at /usr/src/debug/rspamd-1.6.1/src/rspamd.c:607
#19 0x000000000043c2f9 in main (argc=1, argv=0x7ffe14064cf8, env=<optimized out>) at /usr/src/debug/rspamd-1.6.1/src/rspamd.c:1380
```

**Comment by vstakhov** [comment]
You need to disable ARC module when you disable DKIM module.

**Comment by flowerysong** [comment]
Should be fixed by fd6f2fa4115fafd6330297201c06ef2aa6abebb3
